q2_amr/types/_transformer.py

```diff
@@ -233,9 +233,41 @@ def read_mapping_data(data_path, variant):
 
 @plugin.register_transformer
 def _12(data: CARDAlleleAnnotationDirectoryFormat) -> qiime2.Metadata:
-    return read_mapping_data(data, "allele")
+    return tabulate_data(data, "allele")
 
 
 @plugin.register_transformer
 def _13(data: CARDGeneAnnotationDirectoryFormat) -> qiime2.Metadata:
-    return read_mapping_data(data, "gene")
+    return tabulate_data(data, "gene")
+
+
+@plugin.register_transformer
+def _14(data: CARDAnnotationDirectoryFormat) -> qiime2.Metadata:
+    return tabulate_data(data, "mags")
+
+
+def tabulate_data(data_path, data):
```

Comment:
Can you change `data` to something like `data_type`? It's not really obvious what "data" should contain otherwise...